lib/tiled/properties.rb

```diff
@@ -20,10 +20,10 @@ def initialize(map)
     # @param value [Integer, Float, Boolean, String] value of added property
     # @return [Tiled::Properties] self
     def add(name, type = 'string', value)
+      name = name.downcase.gsub(' ', '_')
+
       instance_variable_set("@#{name}", value)
-      instance_variable_set("@#{name}_type", type)
```

Comment:
Tiled supports custom property types, so it is essential to store this field:
```xml
<properties>
  <property name="some file" type="file" value="test_world.world"/>
  <property name="test_class" type="class" propertytype="some_class">
    <properties>
      <property name="member1" value="1234"/>
    </properties>
  </properties>
</properties>
 ```

For the `file` property this info will be lost, it becomes just a string, but for example, for a map, I would want to preload all `file` properties.

Nested properties are not supported yet, but it is not the point :D
Also, if Tiled adds support for new properties type in the future, even with the old version of the library user will be able to distinguish these properties from strings.

Reply:
Ah, good point. Fixed!